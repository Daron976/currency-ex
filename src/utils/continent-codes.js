export const europe = [
  'all', 'amd', 'azn', 'bam', 'bgn', 'byn', 'chf', 'czk', 'dkk', 'eur',
  'gbp', 'gel', 'ggp', 'gip', 'hrk', 'huf', 'imp', 'isk', 'jep', 'kzt',
  'mdl', 'mkd', 'nok', 'pln', 'ron', 'rsd', 'rub', 'sek', 'try', 'uah',
];

export const africa = [
  'aoa', 'bif', 'bwp', 'cdf', 'cve', 'djf', 'dzd', 'egp', 'ern', 'etb',
  'ghs', 'gmd', 'gnf', 'kes', 'kmf', 'lrd', 'lsl', 'lyd', 'mad', 'mga',
  'mru', 'mur', 'mwk', 'mzn', 'nad', 'ngn', 'rwf', 'scr', 'sdg', 'shp',
  'sll', 'sos', 'ssp', 'stn', 'szl', 'tnd', 'tzs', 'ugx', 'xaf', 'xof', 'zar', 'zmw',
];

export const asia = [
  'aed', 'afn', 'amd', 'azn', 'bdt', 'bhd', 'bnd', 'btn', 'cny', 'gel',
  'hkd', 'idr', 'ils', 'inr', 'iqd', 'irr', 'jod', 'jpy', 'kgs', 'khr', 'kpw',
  'krw', 'kwd', 'kzt', 'lak', 'lbp', 'lkr', 'mmk', 'mnt', 'mop', 'mvr', 'myr',
  'npr', 'omr', 'php', 'pkr', 'qar', 'sar', 'sgd', 'syp', 'thb', 'tjs', 'tmt',
  'try', 'twd', 'uzs', 'vnd', 'yer',
];

export const northAmerica = [
  'ang', 'awg', 'bbd', 'bmd', 'bsd', 'bzd', 'cad', 'crc',
  'cup', 'dkk', 'dop', 'gtq', 'hnl', 'htg', 'jmd', 'kyd', 'mxn', 'nio', 'ttd', 'usd', 'xcd'];

export const southAmerica = [
  'ars', 'bob', 'brl', 'clp', 'cop', 'fkp', 'gyd', 'pen', 'pyg', 'srd', 'uyu', 'ves',
];

export const australiaOceana = [
  'aud', 'fjd', 'nzd', 'pgk', 'sbd', 'top', 'vuv', 'wst', 'xpf',
];

export const crypto = [
  'ada', 'algo', 'aur', 'avax', 'bch', 'btc ', 'ckb', 'dash',
  'deso', 'doge', 'dot', 'eos', 'etc', 'eth', 'firo', 'grc',
  'icp', 'ltc', 'mzc', 'nano', 'neo', 'nmc', 'nxt', 'ppc',
  'safemoon', 'shib', 'tit', 'trx', 'usdt', 'vtc', 'xbt',
  'xdg', 'xlm', 'xmr', 'xpm', 'xrp', 'xvg', 'zec',
];
