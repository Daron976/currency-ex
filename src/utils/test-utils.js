/*eslint-disable*/
import React from 'react';
import { render } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import countryReducer from '../redux/country/country';
import currencyReducer from '../redux/currency/curency';
import baseCurrencyReducer from '../redux/conversion/baseCurrency';

export const renderState = (
  ui,
  {
    preloadedState = {},
    store = configureStore({ reducer: { 
      countryReducer,
      currencyReducer,
      baseCurrencyReducer,
     }, preloadedState }),
    ...renderOptions
  } = {},
) => {
  const Wrapper = ({ children }) => <Provider store={store}>{ children }</Provider>;

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};
