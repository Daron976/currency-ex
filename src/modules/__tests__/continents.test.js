/*eslint-disable*/
import { fireEvent, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { renderState } from '../../utils/test-utils';
import Continents from '../continents';

describe('Continents', () => {
  test('Continents items renders as expected', async () => {
    const snapContinents = renderState(
      <BrowserRouter>
        <Continents />
      </BrowserRouter>,
    );

    expect(snapContinents).toMatchSnapshot();
  });
  test('Africa tile is present', async () => {
    renderState(
      <BrowserRouter>
        <Continents />
      </BrowserRouter>,
    );

    const africa = screen.getByText('Africa');

    expect(africa).toBeInTheDocument; 
  });
  test('Africa tile is present', async () => {
    renderState(
      <BrowserRouter>
        <Continents />
      </BrowserRouter>,
    );

    const africa = screen.getByText('Africa');

    fireEvent.click(africa);

    expect(africa).not.toBeInTheDocument;
  });
});
