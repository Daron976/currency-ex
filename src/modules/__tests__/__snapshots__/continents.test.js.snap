// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continents Continents items renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="header"
        >
          <i
            class="fa-solid fa-backward-step"
          />
          <img
            alt="App logo"
            src="https://img.icons8.com/external-icons-smashing-stocks/68/000000/external-Currencies-stock-market-icons-smashing-stocks.png"
          />
          <a
            href="/continents"
          >
            <i
              class="fa-solid fa-house"
            />
          </a>
        </header>
        <section
          class="continents-page"
        >
          <div
            class="intro"
          >
            <div
              class="intro-content"
            >
              <div
                class="intro-details"
              >
                <h1
                  class="intro-text"
                >
                  Currex
                </h1>
                <p
                  class="intro-subtext"
                >
                  Foreign Exchange
                  <br />
                  Latest currency figures
                </p>
              </div>
              <div
                class="scroll"
                style="display: block;"
              >
                Scroll Down
                <br />
                <i
                  class="fa-sharp fa-solid fa-chevron-down fa-3x"
                />
              </div>
            </div>
          </div>
          <section
            class="categories"
          >
            <h2>
              Regions
            </h2>
            <div
              class="categories-container"
            >
              <div
                class="wrapper1"
              >
                <a
                  class="regions"
                  href="/Africa"
                >
                  <div
                    class="category"
                    id="Africa"
                  >
                    <img
                      alt="Africa"
                      src="https://img.icons8.com/3d-fluency/94/null/globe-africa.png"
                    />
                    <p>
                      Africa
                    </p>
                  </div>
                </a>
                <a
                  class="regions"
                  href="/Europe"
                >
                  <div
                    class="category"
                    id="Europe"
                  >
                    <img
                      alt="Europe"
                      src="https://img.icons8.com/3d-fluency/94/null/europe.png"
                    />
                    <p>
                      Europe
                    </p>
                  </div>
                </a>
                <a
                  class="regions"
                  href="/Americas"
                >
                  <div
                    class="category"
                    id="Americas"
                  >
                    <img
                      alt="Americas"
                      src="https://img.icons8.com/3d-fluency/94/null/america.png"
                    />
                    <p>
                      Americas
                    </p>
                  </div>
                </a>
              </div>
              <div
                class="wrapper2"
              >
                <a
                  class="regions"
                  href="/Asia"
                >
                  <div
                    class="category"
                    id="Asia"
                  >
                    <img
                      alt="Asia"
                      src="https://img.icons8.com/3d-fluency/94/null/asia.png"
                    />
                    <p>
                      Asia
                    </p>
                  </div>
                </a>
                <a
                  class="regions"
                  href="/Oceania"
                >
                  <div
                    class="category"
                    id="Oceania"
                  >
                    <img
                      alt="Oceania"
                      src="https://img.icons8.com/3d-fluency/94/null/asia.png"
                    />
                    <p>
                      Oceania
                    </p>
                  </div>
                </a>
              </div>
            </div>
          </section>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="header"
      >
        <i
          class="fa-solid fa-backward-step"
        />
        <img
          alt="App logo"
          src="https://img.icons8.com/external-icons-smashing-stocks/68/000000/external-Currencies-stock-market-icons-smashing-stocks.png"
        />
        <a
          href="/continents"
        >
          <i
            class="fa-solid fa-house"
          />
        </a>
      </header>
      <section
        class="continents-page"
      >
        <div
          class="intro"
        >
          <div
            class="intro-content"
          >
            <div
              class="intro-details"
            >
              <h1
                class="intro-text"
              >
                Currex
              </h1>
              <p
                class="intro-subtext"
              >
                Foreign Exchange
                <br />
                Latest currency figures
              </p>
            </div>
            <div
              class="scroll"
              style="display: block;"
            >
              Scroll Down
              <br />
              <i
                class="fa-sharp fa-solid fa-chevron-down fa-3x"
              />
            </div>
          </div>
        </div>
        <section
          class="categories"
        >
          <h2>
            Regions
          </h2>
          <div
            class="categories-container"
          >
            <div
              class="wrapper1"
            >
              <a
                class="regions"
                href="/Africa"
              >
                <div
                  class="category"
                  id="Africa"
                >
                  <img
                    alt="Africa"
                    src="https://img.icons8.com/3d-fluency/94/null/globe-africa.png"
                  />
                  <p>
                    Africa
                  </p>
                </div>
              </a>
              <a
                class="regions"
                href="/Europe"
              >
                <div
                  class="category"
                  id="Europe"
                >
                  <img
                    alt="Europe"
                    src="https://img.icons8.com/3d-fluency/94/null/europe.png"
                  />
                  <p>
                    Europe
                  </p>
                </div>
              </a>
              <a
                class="regions"
                href="/Americas"
              >
                <div
                  class="category"
                  id="Americas"
                >
                  <img
                    alt="Americas"
                    src="https://img.icons8.com/3d-fluency/94/null/america.png"
                  />
                  <p>
                    Americas
                  </p>
                </div>
              </a>
            </div>
            <div
              class="wrapper2"
            >
              <a
                class="regions"
                href="/Asia"
              >
                <div
                  class="category"
                  id="Asia"
                >
                  <img
                    alt="Asia"
                    src="https://img.icons8.com/3d-fluency/94/null/asia.png"
                  />
                  <p>
                    Asia
                  </p>
                </div>
              </a>
              <a
                class="regions"
                href="/Oceania"
              >
                <div
                  class="category"
                  id="Oceania"
                >
                  <img
                    alt="Oceania"
                    src="https://img.icons8.com/3d-fluency/94/null/asia.png"
                  />
                  <p>
                    Oceania
                  </p>
                </div>
              </a>
            </div>
          </div>
        </section>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
